components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/impl/EndpointRegistryImpl.java

```diff
@@ -113,24 +119,35 @@ public void deleteEndpointRegistry(String registryUUID) throws EndpointRegistryE
     /**
      * Returns details of all Endpoint Registries belong to a given tenant
      *
-     * @param name         Registry name
-     * @param sortBy       Name of the sorting field
-     * @param sortOrder    Order of sorting (asc or desc)
-     * @param limit        Limit
-     * @param offset       Offset
      * @param tenantDomain
      * @return A list of EndpointRegistryInfo objects
      * @throws EndpointRegistryException if failed to get details of an Endpoint Registries
      */
-    public List<EndpointRegistryInfo> getEndpointRegistries(String name, String sortBy, String sortOrder,
-                                                            int limit, int offset,
-                                                            String tenantDomain) throws EndpointRegistryException {
+    public List<EndpointRegistryInfo> getEndpointRegistries(String tenantDomain) throws EndpointRegistryException {
 
         int tenantId;
         try {
             tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
                     .getTenantId(tenantDomain);
-            return registryDAO.getEndpointRegistries(name, sortBy, sortOrder, limit, offset, tenantId);
+            // Create the default registry if no registry found
+            if (!registryDAO.isEndpointRegistryTypeExists(EndpointRegistryConstants.REGISTRY_TYPE_WSO2, tenantId) &&
```

Comment:
Rather than checking for the existence and then creating the Registry, can't we simply fetch the registry and then create it if it's null?

Reply:
Fixed